### ISSUE TITLE
Exception when slicing array of length one with True boolean list/array

### ISSUE THREAD

**Issue by freidrichen** [dask/dask#3290]
Repro:
```python
import dask.array as da

d = da.ones(1, chunks=(5,))
d[[True]]
```

Result:
```
Traceback (most recent call last):
  File "dask_mcve.py", line 6, in <module>
    d[[True]]
  File "python3.6/site-packages/dask/array/core.py", line 1299, in __getitem__
    index2 = normalize_index(index, self.shape)
  File "python3.6/site-packages/dask/array/slicing.py", line 775, in normalize_index
    check_index(i, d)
  File "python3.6/site-packages/dask/array/slicing.py", line 818, in check_index
    raise IndexError("Index out of bounds %s" % dimension)
IndexError: Index out of bounds 1
```

When slicing with `[False]` instead of `[True]` it works as expected. I also get the same results when slicing with the equivalent numpy arrays.

Edit:
Python version: 3.6.4
Dask version: 0.17.1
Numpy version: 1.14.0

**Comment by jakirkham** [comment]
IIUC we don't actually support slicing with a `list` of `bool`s. Though think it should be possible given we already support a `list` of `int`s.

In the short run, you could convert a `list` of `bool`s into a `list` of `int`s.

Would be open to a PR if you would be interested.

**Comment by freidrichen** [comment]
The documentation states that lists of bools are supported: http://dask.pydata.org/en/latest/array-slicing.html

If I can find the time, I might look into creating a PR.

**Comment by ZiyaoWei** [comment]
Just tried this again and seems to be fixed by #3728 - @freidrichen can you confirm? Sorry if I have scooped you :-)

**Comment by freidrichen** [comment]
Yes, this seems to work now. Thanks for fixing this! =)
